rollup.config.js

```diff
@@ -44,6 +44,43 @@ function generateCssModulesMockName(name) {
   return name;
 }
 
+function handleNextEntry() {
+  return {
+    name: "handle-next-entry",
+    writeBundle() {
+      const nextFolder = path.join(DIST_PATH, "esm", "next");
+
+      // Ensure the 'next' directory exists.
+      if (!fs.existsSync(nextFolder)) {
+        fs.mkdirSync(nextFolder);
+      }
+
+      // Paths
+      const originalJsPath = path.join(DIST_PATH, "esm", "src", "next.js");
+      const originalDtsPath = path.join(DIST_PATH, "esm", "next.d.ts");
+      const targetJsPath = path.join(nextFolder, "next.js");
+      const targetDtsPath = path.join(nextFolder, "next.d.ts");
+
+      // Move JS and DTS files
+      if (fs.existsSync(originalJsPath)) {
+        fs.renameSync(originalJsPath, targetJsPath);
+      }
+      if (fs.existsSync(originalDtsPath)) {
+        fs.renameSync(originalDtsPath, targetDtsPath);
+      }
+
+      // Create package.json in the 'next' folder
+      const packageData = {
```

Comment:
tried it as a soultion for types (for autocomplete but not worked, will remove).
the rest of the plugin is for moving the d.ts and js file to one location which compiled by tsconfig (originally next.js was under esm which was problematic)